dll/win32/syssetup/wizard.c

```diff
@@ -2992,6 +3020,12 @@ ProcessSetupInf(
                              (LPBYTE)pSetupData->SourcePath,
                              (wcslen(pSetupData->SourcePath) + 1) * sizeof(WCHAR));
 
+        res = RegSetValueExW(hKey,
+                             L"Installation Sources",
+                             0,
+                             REG_MULTI_SZ,
+                             (LPBYTE)szValue,
+                             (DWORD)((pData - szValue) * sizeof(WCHAR)));
```

Comment:
???
```suggestion
                             (DWORD)((pData - szValue + 1) * sizeof(WCHAR)));
```

Reply:
Why add +1 ?
That has been done above already? (`pData += cchLength + 1;`)

Reply:
Double NUL is needed.

Reply:
First NULL is added on string copy:
```c
    hr = StringCchCopyW(pData, cchLength, pSetupData->SourcePath);
```
Second NULL is added manually later:
```c
    pData += cchLength + 1;
    *pData = UNICODE_NULL;
```
Am I right?

Reply:
The interval `[0, pData - szValue)` doesn't contain the last NUL.

Reply:
https://en.m.wikipedia.org/wiki/Interval_(mathematics)